components/plan-build/bin/hab-plan-build.sh

```diff
@@ -1569,7 +1567,15 @@ do_download() {
 
 # Default implementation for the `do_download()` phase.
 do_default_download() {
-  download_file $pkg_source $pkg_filename $pkg_shasum
+  if [[ $pkg_source == "."* ]]; then
+    # File is a local path, so is already downloaded.
+    cp -r $pkg_source $HAB_CACHE_SRC_PATH/$pkg_filename
```

Comment:
@Hoverbear What do you think about putting the copy into a different function such as `do_source_copy` or `do_local_copy` and calling the function within this `if` statement?

My hope is rather than telling people to write a `do_download` callback to customize local file copies, this would allow us to provide a more logically named callback for users should they want to change the copy command or behavior. Do you think that makes sense or would end up being a more confusing user experience?